src/lib/run-ggshield.ts

```diff
@@ -12,38 +16,32 @@ import * as os from "os";
  * @returns
  */
 export function runGGShieldCommand(
-    configuration: GGShieldConfiguration,
-    args: string[]
-  ): SpawnSyncReturns<string> {
-    const { ggshieldPath, apiUrl, apiKey } = configuration;
-    let env: {
-      GITGUARDIAN_API_URL: string;
-      GG_USER_AGENT: string;
-      GITGUARDIAN_API_KEY?: string;
-      } = {
-      GITGUARDIAN_API_URL: apiUrl,
-      GG_USER_AGENT: "gitguardian-vscode",
-    };
+  configuration: GGShieldConfiguration,
+  args: string[]
+): SpawnSyncReturns<string> {
+  const { ggshieldPath, apiUrl, apiKey } = configuration;
+  let env: NodeJS.ProcessEnv = {
+    ...process.env,
+    GITGUARDIAN_API_URL: apiUrl,
+    GG_USER_AGENT: "gitguardian-vscode",
+  };
 
-    if (apiKey) {
-      env = {
-        ...env,
-        GITGUARDIAN_API_KEY: apiKey,
-      };
-    }
-
-    let options: SpawnSyncOptionsWithStringEncoding = {
-      cwd: os.tmpdir(),
-      env: env,
-      encoding: "utf-8",
-      windowsHide: true,
-    };
+  if (apiKey) {
+    env.GITGUARDIAN_API_KEY = apiKey;
+  }
 
-    // If the command is executed in a worskpace, execute ggshield from the root folder so .gitguardian.yaml is used
-    if (workspace.workspaceFolders?.length || 0 > 0) {
-      options["cwd"] = workspace.workspaceFolders![0].uri.fsPath;
-    }
-    let proc = spawnSync(ggshieldPath, args, options);
+  let options: SpawnSyncOptionsWithStringEncoding = {
+    cwd: os.tmpdir(),
```

Comment:
are we required to have a `cwd` ? Could we not just omit it ?

Reply:
I would prefer to keep it, I think it's a safer option, especially for cross-platform compatibility. I'm not sure of how `cwd` would work in different platform if the user opens a single file and not a folder so I'd prefer we keep it.